### PR TITLE
Short documentation for datetime64

### PR DESCRIPTION
**Pull request by jl-wynen**

### CODE REVIEW

File: docs/about/release-notes.rst [hunk 1]
```diff
@@ -23,6 +23,11 @@ Features
   * Support loading/converting Mantid ``WorkspaceGroup``, this will produce a ``dict`` of data arrays `#1654 <https://github.com/scipp/scipp/pull/1654>`_.
   * Fixes to support loading/converting ``McStasNexus`` files `#1659 <https://github.com/scipp/scipp/pull/1659>`_.
 
+* Support for datetime
+
+  * ``sc.dtype.datetime64`` with copy-less casting between numpy and and scipp where possible. `#1639 <https://github.com/scipp/scipp/pull/1639>`_
```
Comment (SimonHeybrock) [inline docs/about/release-notes.rst:28]:
```suggestion
  * ``sc.dtype.datetime64`` with copy-less casting between numpy and scipp where possible. `#1639 <https://github.com/scipp/scipp/pull/1639>`_
```

File: docs/python-reference/dtype.ipynb [hunk 2]
```diff
@@ -6,8 +6,7 @@
    "source": [
     "# Data types\n",
     "\n",
-    "Scipp supports the most common data types (`dtype`).\n",
-    "In most cases the `dtype` is derived from the data, e.g., when passing a [numpy.ndarray](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.html#numpy.ndarray) to scipp, scipp will use `dtype` provided by numpy:"
+    "In most cases, the data type (`dtype`) of a [Variable](../generated/scipp.Variable.html) is derived from the data. For instance when passing a [numpy array](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.html#numpy.ndarray) to scipp, scipp will use the `dtype` provided by numpy:"
```
Comment (SimonHeybrock) [inline docs/python-reference/dtype.ipynb:9]: We are generally trying to put one sentencce per line in docs/markdown to improve diff readability.

File: docs/python-reference/dtype.ipynb [hunk 3]
```diff
@@ -38,7 +37,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The `dtype` may also be specified using a keyword argument, both [numpy.dtype](https://docs.scipy.org/doc/numpy/reference/generated/numpy.dtype.html#numpy.dtype) and `scipp.dtype` are supported:"
+    "The `dtype` may also be specified using a keyword argument. It is possible to use scipp's own`scipp.dtype`, [numpy.dtype](https://docs.scipy.org/doc/numpy/reference/generated/numpy.dtype.html#numpy.dtype), or a string:"
```
Comment (SimonHeybrock) [inline docs/python-reference/dtype.ipynb:40]: space after `own`.

File: docs/python-reference/dtype.ipynb [hunk 4]
```diff
@@ -82,8 +91,163 @@
...
+   "outputs": [],
+   "source": [
+    "sc.scalar(value=681794055, unit=sc.units.s, dtype=sc.dtype.datetime64)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Datetime Variables always need a temporal unit and that unit determines how the integer that is passed to `value=` is interpreted:"
```
Comment (SimonHeybrock) [inline docs/python-reference/dtype.ipynb:142]: "variables" (lowercase)

File: docs/python-reference/dtype.ipynb (hunk 3 above)
Comment (OwenArnold) [inline docs/python-reference/dtype.ipynb:40]:
```suggestion
    "The `dtype` may also be specified using a keyword argument. It is possible to use scipp's own `scipp.dtype`, [numpy.dtype](https://docs.scipy.org/doc/numpy/reference/generated/numpy.dtype.html#numpy.dtype), or a string:"
```

File: docs/python-reference/dtype.ipynb [hunk 5]
```diff
@@ -82,8 +91,163 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.scalar(value=np.datetime64('2021-09-03T12:30:00'))"
+   ]
```
Comment (SimonHeybrock) [inline docs/python-reference/dtype.ipynb:250]: Is it worth making a separate page for datetime?

File: docs/python-reference/dtype.ipynb [hunk 6]
```diff
@@ -82,8 +91,163 @@
...
+    "Datetime Variables always need a temporal unit and that unit determines how the integer that is passed to `value=` is interpreted:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var = sc.scalar(value=681794055, unit=sc.units.ns, dtype=sc.dtype.datetime64)\n",
```
Comment (OwenArnold) [inline docs/python-reference/dtype.ipynb:151]: Is it relevant to say (or show) what happens if a non-time unit is used?

**Review by OwenArnold — COMMENTED** [review]
Looks good. Is there any developer level documentation it's worth capturing alongside your various changes?

**Comment by SimonHeybrock** [comment]
> Looks good. Is there any developer level documentation it's worth capturing alongside your various changes?

Do you have something conrete in mind you would want documented here? Aren't code comments and doxygen (where applicable) enough for most of this?

**Comment by OwenArnold** [comment]
> Do you have something conrete in mind you would want documented here? Aren't code comments and doxygen (where applicable) enough for most of this?

I assumed so. But I'm raising the question anyway as this is easy to overlook.

File: docs/python-reference/dtype.ipynb (hunk 6 above)
Comment (jl-wynen) [reply docs/python-reference/dtype.ipynb:151]: Not sure. I generally prefer avoiding showing how not to do things. Is it useful to know that you can swap out the unit and effectively break the variable?

**Comment by jl-wynen** [comment]
The bindings code is scattered all over the place. Not sure if there is much to document in a central place. And the underlying implementation is not complicated and has no particular design that one could explain.

Maybe it would be useful to write something about the bindings in general. But there are so many special cases that it seems difficult to summarise.

File: docs/python-reference/dtype.ipynb (hunk 5 above)
Comment (jl-wynen) [reply docs/python-reference/dtype.ipynb:252]: Hm, maybe. It is quite short at the moment

**Review by OwenArnold — APPROVED** [review]

**Comment by OwenArnold** [comment]
@SimonHeybrock Did you have further comments? Can we merge this?

**Pull request merged**